src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -384,7 +397,13 @@ static UMessage toMessage(CloudEvent event) {
         if (hasCommunicationStatusProblem(event)) {
             builder.setCommstatus(getCommunicationStatus(event));
         }
-        getPriority(event).map(p -> p.startsWith("UPRIORITY_") ? p : "UPRIORITY_" + p).map(UPriority::valueOf).ifPresent(builder::setPriority);
+        getPriority(event).map(p -> UPriority.getDescriptor().getValues().stream()
+                .filter(v -> v.getOptions().hasExtension(UprotocolOptions.ceName) &&
+                v.getOptions().getExtension(UprotocolOptions.ceName).equals(p))
+                .map(v -> UPriority.forNumber(v.getNumber()))
+                .findFirst()
+                .orElse(UPriority.UNRECOGNIZED)
```

Comment:
so if it can't be found you use UNRECOGNIZED and then ifPresent, so we get UNRECOGNIZED if missing, would it not be better to not have the orElse and then the ifPresent will not be called and the builder will not set any priority.
Having the UNRECOGNIZED feels like it could lead to all kinds of settings that are not needed in this case

Reply:
@tamarafischer you bring up an important point actually because priority is mandatory and If I do not set, protobuf will assign it to UNRECOGNIZED when we build UAttributes object so it is better to set it to UPRIORITY_UNSPECIFIED and then add this check in the validator.